tests/test_packaging_guardrails.py

```
@@ -0,0 +1,126 @@
+    assert re.search(r"^Name\s*=\s*.+$", content, re.MULTILINE)
+    assert re.search(r"^Exec\s*=\s*.+$", content, re.MULTILINE)
+    assert re.search(r"^Icon\s*=\s*.+$", content, re.MULTILINE)
+    assert re.search(r"^Type\s*=\s*Application$", content, re.MULTILINE)
+
+
+def test_linux_appdata_xml_validity():
+    """Verify packaging/linux/immich-go-gui.appdata.xml is well-formed XML with valid ID."""
+    appdata_path = LINUX_PKG_ROOT / "immich-go-gui.appdata.xml"
+    assert appdata_path.is_file(), f"Missing {appdata_path}"
+
+    tree = ET.parse(appdata_path)
```

Ruff (0.16.1)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)